src/main/java/com/github/feroult/gapi/spreadsheet/SpreadsheetAPIImpl.java

```diff
@@ -226,18 +226,21 @@ private void adjustWorksheetDimensions(SpreadsheetBatch updateSet, BatchOptions.
 		}
 
 		List<List<Object>> body = response.getValues();
-		List<Object> header = body.get(0);
+		if (body == null) {
```

Comment:
It's great to fix it to maintain compatibility. That being said, the behaviour before was terrible. This normalization thing is unexpected, brittle and source of many bugs. My suggestion is to make these changes to asMap, as you did, but deprecate asMap and immediately introduce a new method, choose a name, like asMapV2 or something that does not normalize stuff. That will encourage clients to upgrade to the new version and hopefully remove their codes that deal with the "expected" normalization.

Reply:
I did a second method called `asMapWithoutNormalization`, but I don't know if it's necessary to deprecate the `asMap` method, because now it's gonna be possible to choose between a method that normalize or not.